realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -537,9 +470,12 @@ public void onChange(Realm object) {
                     realm.removeAllChangeListeners();
                     realm.addChangeListener(this);
                     realm.addChangeListener(listenerA);
-                    realm.beginTransaction();
-                    realm.commitTransaction();
-
+                    // Async transaction to avoid endless recursion.
+                    realm.executeTransactionAsync(new Realm.Transaction() {
```

Comment:
Yeah this one looks a little scary to me too, although it is to be expected if commit calling `RealmChangeListener` on looper thread results in immediately calling `notifyChangeListeners()`

I always wondered why it was delegated to the next event loop in 0.88.0, I was looking for it but haven't found anything conclusive.